lib/geo_pattern/pattern_db.rb

```diff
@@ -0,0 +1,82 @@
+module GeoPattern
+  class PatternDb
+    private
+
+    attr_reader :patterns
+
+    public
+
+    def initialize
+      @patterns = {
+        'chevrons'            => ChevronPattern,
+        'concentric_circles'  => ConcentricCirclesPattern,
+        'diamonds'            => DiamondPattern,
+        'hexagons'            => HexagonPattern,
+        'mosaic_squares'      => MosaicSquaresPattern,
+        'nested_squares'      => NestedSquaresPattern,
+        'octagons'            => OctagonPattern,
+        'overlapping_circles' => OverlappingCirclesPattern,
+        'overlapping_rings'   => OverlappingRingsPattern,
+        'plaid'               => PlaidPattern,
+        'plus_signs'          => PlusSignPattern,
+        'sine_waves'          => SineWavePattern,
+        'squares'             => SquarePattern,
+        'tessellation'        => TessellationPattern,
+        'triangles'           => TrianglePattern,
+        'xes'                 => XesPattern
+      }.freeze
+    end
+
+    def count
+      patterns.count
+    end
+
+    def all
+      patterns.values
+    end
+
+    def fetch(*requested_patterns)
+      requested_patterns = requested_patterns.flatten.compact
+
+      return all if requested_patterns.empty?
+
+      output_warning = false
+      fail InvalidPatternError, "Error: At least one of the requested patterns \"#{requested_patterns.join(", ")}\" is invalid" unless valid?(requested_patterns)
+
+      result = requested_patterns.map do |pattern|
+        if pattern.kind_of? String
+          output_warning = true
+
+          patterns[pattern]
+        else
+          pattern
+        end
+      end
+
+      puts SVG.as_comment("String pattern references are deprecated as of 1.3.0") if output_warning
```

Comment:
The deprecation warning is output w/ the SVG string as a comment so that the resulting SVG is still valid. I'm not sure many people will see the warning this way, but this seems like the best way since this is the only output.